contributing/core/testing.md

```diff
@@ -0,0 +1,64 @@
+# Testing
+
+## Running tests
+
+We recommend running the tests in headless mode (with the browser windows hidden) and with a specific directory pattern and/or test name (`-t`) which ensures only a small part of the test suite is run locally:
+
+For example, running one test in the production test suite:
+
+````
+Running one test in the `test/integration/production` test suite:
+
+```sh
+pnpm test -- test/integration/production -t "should allow etag header support"
```

Comment:
We need `--` to pass the args to jest correctly. See: https://github.com/pnpm/pnpm/issues/4821

Reply:
Wow, I didn't notice that before!

I always use commands like this:

```sh
pnpm run testonly --test-path-pattern test/integration/script-loader/test/index.test.js
```

And pnpm is able to forward `--test-path-pattern` to jest.

Reply:
This seems to work correctly as long as the command isn't `test` literally so lets just avoid recommending that one and use the explicit `testheadless` by default